src/session.ts

```diff
@@ -344,4 +359,19 @@ function getReturnPathname(url: string): string {
   return `${newUrl.pathname}${newUrl.searchParams.size > 0 ? '?' + newUrl.searchParams.toString() : ''}`;
 }
 
+function getScreenHint(signUpPaths: string[] | string | undefined, pathname: string) {
+  if (!signUpPaths) return 'sign-in';
+
+  if (!Array.isArray(signUpPaths)) {
+    return signUpPaths.includes(pathname) ? 'sign-up' : 'sign-in';
+  }
+
+  const screenHintPaths: string[] = signUpPaths.filter((pathGlob) => {
+    const pathRegex = getMiddlewareAuthPathRegex(pathGlob);
+    return pathRegex.exec(pathname);
+  });
```

Comment:
Is there a reason we pass the paths through `getMiddlewareAuthPathRegex` if `signUpPaths` is an array, but not if it's a single value? Could a single value also be a path glob?

Reply:
That's an excellent point!